react/tic-tac-toe/src/app/components/AuthRoute/index.js

```diff
@@ -0,0 +1,42 @@
+import React, { Component } from 'react';
+import { Route } from 'react-router-dom';
+import { push } from 'connected-react-router';
+import { connect } from 'react-redux';
+import PropTypes from 'prop-types';
+
+import { LOGIN } from '../../../constants/routes';
+import { TOKEN } from '../../../constants/auth';
+
+class AuthRoute extends Component {
+  componentDidMount() {
+    if (this.props.isPrivate) {
+      const token = localStorage.getItem(TOKEN);
```

Comment:
You are accessing the local storage each time you change the current screen. This should be accessed once (on app mount) and saved in the store, and then used from there